flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroup.java

```diff
@@ -441,14 +441,14 @@ private MetricGroup addGroup(String name, boolean keyed) {
 			}
 			else {
 				// return a non-registered group that is immediately closed already
-				GenericMetricGroup closedGroup = createChildGroup(name);
+				GenericMetricGroup closedGroup = createChildGroupInKeyedContext(name);
```

Comment:
Lets return a GenericMetricGroup here to keep this case as simple as possible.

Reply:
Have addressed it.